tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts.py

```diff
@@ -1389,7 +1389,7 @@ def bad_request(*args, **kwargs):
     monkeypatch.setattr(S3Handler, "_etag_from_obj", bad_request)
 
     with pytest.raises(FileNotFoundError, match="Unable to find"):
-        artifact.download()
+        artifact._download()
```

Comment:
I'd rather test the user-facing methods. Why did you change this? Same below.

Reply:
Agreed. I shouldn't have done that. One question: We're monkeypatching the Aws client method to return 404 if we try to access a file. Could the fact that this is happening in the background process now be causing issues? I'm not seeing the same errors while running the tests, hence the results are not as expected.